sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images/sycl_ext_oneapi_bindless_images.asciidoc

```diff
@@ -0,0 +1,1826 @@
+image memory allocation such as the range, channel type, channel order, number 
+of channels, number of levels, and image type.
+
+In the case where a mipmap has been allocated, `get_mip_level` can be used to 
+return an `image_mem_handle` to a specific level of the mipmap. This can then 
+be used to copy data to that specific level or create an image handle based on 
+that level.
+
+Note that the handle type `image_mem_handle::raw_handle_type` is an opaque type, 
+and the handle cannot be dereferenced on the host. The layout of the memory is 
+backend-specific, and may be an optimized layout, e.g. tile swizzle patterns.
+
+`image_mem` shall not be copy-constructible, copy-assignable, or 
+device-copyable.
+`image_mem` shall be default-constructible, move-constructible, as well as 
+provide a move-assignment operator.
+
+`image_mem_handle` shall be default-constructible and copy-constructible.
+`image_mem_handle` shall not be device-copyable.
+
+If the construction or destruction of the `image_mem` class fails, a 
+`sycl::exception` with error code `sycl::errc::memory_allocation` will be 
+thrown.
+
+Similarly, if `alloc_image_mem` or `free_image_mem` fail, a `sycl::exception` 
+with error code `sycl::errc::memory_allocation` will be thrown.
+
+[NOTE]
+====
+In the DPC++ CUDA backend, `image_mem` will allocate/deallocate a 
+`CUarray` type (or `CUmipmappedArray` in the case of mipmap images).
+====
+
+===== Getting image information from `image_mem_handle`
+
+Extension functions are provided to retrieve information about images allocated 
+using the `image_mem_alloc` function. These are similar to the member functions 
+provided by `image_mem`. However, since the `image_mem_handle` is a minimal 
+struct representing just the opaque handle the underlying memory object, there 
+is some information that we cannot retrieve from it, namely the `image_type`,
+`image_channel_order`, the `sycl::context` the memory was allocated in, and the 
+`image_descriptor` used to allocate the memory.
+
+```cpp
+namespace sycl::ext::oneapi {
+
+sycl::range<3> get_image_range(const sycl::context &syclContext,
+                               const image_mem_handle mem_handle);
+
+sycl::image_channel_type
+get_image_channel_type(const sycl::context &syclContext,
+                       const image_mem_handle mem_handle);
+
+unsigned int get_image_num_channels(const sycl::context &syclContext,
+                                    const image_mem_handle mem_handle);
+}
+```
+
+For `get_image_range` where the underlying image memory was allocated with one 
+or two dimensions, the returned `sycl::range<3>` will contain zero values for 
+the dimensions unused by the underlying image memory object.
+
+==== Allocating USM image memory 
+
+The second way to allocate image memory is to use USM allocations. SYCL already 
+provides a number of USM allocation functions. This proposal would add another,
+pitched memory allocation, through `pitched_alloc_device`.
```

Comment:
I'm not sure if you are entertaining suggestions on names, but I think `malloc_device_image` or `malloc_device_pitched_image` might be better.  That's more consistent with the existing USM allocation function names, which reinforces that these are allocating regular USM device memory.

Reply:
We're entertaining any and all suggestions. The SYCL 2020 spec has USM allocation functions named similarly, e.g. `sycl::aligned_alloc_device`, I think that `pitched_alloc_device` follows that style.

As for including "image" in the name, I'm leaning towards not including it. I don't think pitched allocation is exclusive to images. From my understanding it can be used for regular buffers as well, as its purpose is to align rows of memory to provide better performing accesses. But I might be wrong about this and this benefit does not apply to buffers, only images.